spec/controllers/api/current/appointments_controller_spec.rb

```diff
@@ -34,6 +34,36 @@ def create_record_list(n, options = {})
   describe 'POST sync: send data from device to server;' do
     it_behaves_like 'a working sync controller creating records'
     it_behaves_like 'a working sync controller updating records'
+
+    context 'handles appointment_type correctly' do
+      before :each do
+        set_authentication_headers
+      end
+
+      describe 'stores appointment_type correctly' do
+        let(:request_key) { model.to_s.underscore.pluralize }
+        let(:new_records) { (1..5).map { build_payload.call } }
+        let(:new_records_payload) { Hash[request_key, new_records] }
+
+        it 'creates new records with appointment_type' do
+          before_post_appointments_count = Appointment.all.count
+
+          post(:sync_from_user, params: new_records_payload, as: :json)
+
+          expect(model.count).to eq 5
+          expect(response).to have_http_status(200)
+
+          after_post_appointments = Appointment.all
+          after_post_appointments_ids = after_post_appointments.map(&:id)
+          expect(after_post_appointments.count()).to eq before_post_appointments_count + new_records.count()
```

Comment:
`.count()` => `.count`

Reply:
Removed sync count check as per @govindkrjoshi's comment - the shared specs should take care of checking the sync counts.